### PR TITLE
Fix/active routes push new state to history

### LINKED ISSUE #295: Active routes push new state to history
Just another question :) 

I noticed that active routes push new history states to the browser. For example, if you go to https://reach.tech/router, then click Accessibility, then Credits, and then click Credits 2 or 3 more times, then start hitting the back button, the site will stay on Credits. I think each click is pushing new items onto the history stack.

I'm not sure if this behavior is intentional but I found it surprising, as its not how the browser seems to work on server-rendered sites (even if clicking the active link triggers a full-page refresh). 

Couldn't find anything in the issues about it so figured I'd kick it off here! Thanks for any info.

### PR DESCRIPTION
**Pull request by jzabala**
Based on my comments on the issue: https://github.com/reach/router/issues/295#issuecomment-529122390

Closes #295

### CODE REVIEW

**Comment by blainekasten** [comment]
@jzabala thank you for the great work here and sorting through the requirements! Option 3 was a great way to go :)

**Comment by jzabala** [comment]
> @jzabala thank you for the great work here and sorting through the requirements! Option 3 was a great way to go :)

Awesome. Thanks for taking the time to review the PR @blainekasten .

**Comment by samselikoff** [comment]
Nice job!

File: src/index.js [hunk 1]
```diff
@@ -417,7 +418,15 @@ let Link = forwardRef(({ innerRef, ...props }, ref) => (
                 if (anchorProps.onClick) anchorProps.onClick(event);
                 if (shouldNavigate(event)) {
                   event.preventDefault();
-                  navigate(href, { state, replace });
+                  let shouldReplace = replace;
+                  if (replace !== "boolean" && isCurrent) {
```
Comment (brendanmoore) [inline src/index.js]: `typeof replace !== "boolean"` ?

Comment (jzabala) [reply src/index.js]: I only want to execute the is current replace feat if the user hasn't set the kind of `replace` she wants.

If I just evaluate `replace && isCurrent` and the user says she wants to replace and isCurrent is true, the if is going to evaluate to true, a `shallowCompare` will happen and `shouldReplace` could be false, even though the user is clearly specifying she wants to replace.

So, this feature just kick off if no `replace` options has been set.

Comment (stefanprobst) [reply src/index.js]: i think this is about the missing `typeof`

Comment (jzabala) [reply src/index.js]: Oh thanks @brendanmoore and @stefanprobst for pointing that out. I totally missed. What a silly mistake 😅.

I'll be creating a follow up PR to fix it.

Comment (blainekasten) [reply src/index.js]: Ooof, great catch everyone! Should the unit tests not have caught this?

Comment (blainekasten) [reply src/index.js]: I took care of this, thanks everyone for flagging!

**Pull request merged**